src/tools/clippy/clippy_utils/src/consts.rs

```diff
@@ -778,8 +778,10 @@ pub fn mir_to_const<'tcx>(lcx: &LateContext<'tcx>, result: mir::Const<'tcx>) ->
                 let range = alloc_range(offset + size * idx, size);
                 let val = alloc.read_scalar(&lcx.tcx, range, /* read_provenance */ false).ok()?;
                 res.push(match flt {
+                    FloatTy::F16 => unimplemented!("f16_f128"),
```

Comment:
I'm a bit wary about adding this, especially to tools. There's definitely a risk of forgetting some of them, causing ICEs.

Reply:
I can change this, what would be better?

My original change did more https://github.com/rust-lang/rust/pull/114607/files#diff-83cfffe9dbc2758ecf7b34a433525befc32ebff4525aef0de105fa5e7b659c21R808-R815, but since the surface area of this PR is intentionally much smaller, the primitives aren't available for use.

Reply:
I see the benefit of small PRs and this comment isn't meant to block the PR. Just a concern, that we shouldn't forget about those `unimplemented!` macros.

Reply:
Removing the panics could be added as a step to the tracking issue

Reply:
In fairness to @tgross35, this is impossible to hit right now. I think this approach is the best for the moment.

Reply:
I added a todo list that mentions these at https://github.com/rust-lang/rust/issues/116909#issuecomment-1969554030